book-content/chapters/01-setup-typescript.md

```diff
@@ -0,0 +1,143 @@
+
+When you visit the site, you'll see two options: LTS and Current.
+
+LTS is short for "Long Term Support". This is the recommended version for production use. It's the most stable version, and the one we'll be using in this book.
+
+The Current version contains the latest features, but it's not recommended for production use.
+
+Click the LTS button to download the installer and follow the installation instructions.
+
+After running the installer, you can verify it installed correctly by opening a terminal and running the following command:
+
+```
+node -v
+```
+
+If Node.js is installed correctly, this command will display the version number. If you see an error message like "node command not found," it means the installation was not successful, and you should try again.
+
+### Alternative Package Management with PNPM
+
+Node.js includes the `npm` package manager by default.
+
+If you've worked with JavaScript repositories, you're likely familiar with `npm` and the `package.json` file. The `package.json` file represents all the packages we need to install to run the code in the repository.
+
+For example, in the repository for this material, we have a special CLI for running exercises along with helper packages and various other dependencies like `cross-fetch` and `nodemon`:
+
+```json
+// package.json
+
+{
+  "devDependencies": {
+    "@total-typescript/exercise-cli": "0.4.0",
+    "@total-typescript/helpers": "~0.0.1",
+    "cross-fetch": "~3.1.5",
+    "nodemon": "~3.0.1",
+    "npm-run-all": "~4.1.5",
+    "prettier": "~2.8.7",
+    "typescript": "~5.2.2",
+    "vite-tsconfig-paths": "~4.0.7",
+    "vitest": "0.34.4"
+  }
+}
+```
+
+To install these packages, you would typically run the `npm install` command, which downloads them from the npm registry into your `node_modules` folder. The `node_modules` folder contains JavaScript files that the exercises in the `src` directory need to run.
+
+However, for the book's repository we will be using the PNPM package manager instead.
+
+`pnpm` is used the same way as `npm`, but it is more efficient. Instead of having individual `node_modules` folders for each project, `pnpm` uses a single location on your computer and hard links the dependencies from there. This makes it run faster and use less disk space. I use PNPM in all my projects.
+
+To install PNPM, follow the instructions provided in the [official documentation](https://pnpm.io/installation).
+
+## Installing TypeScript
+
+TypeScript and its dependencies are contained within a single package, called typescript.
+
+You can install it globally with either pnpm or npm:
+
+```
+pnpm add -g typescript
+```
+
+or
+
+```
+npm install –-global typescript
+```
+
+TypeScript is usually also installed in your `package.json` to make sure that all developers using the project are using the same version. For the purposes of this book, a global installation will do just fine.
```

Comment:
It might be worth mentioning here that to use the same version of TS in VS Code you actually have to go through Command Palette and "Select TypeScript version..." to choose the workspace version. Even though it generates a `.vscode/settings.json` file that you could commit... that file is not enough to ensure that other collaborators use the same version of the TS Server. VS Code requires explicit user consent so everybody on the team has to do it manually. It's like that for security reasons